src/main/scala/com/spotify/scio/ScioInjector.scala

```diff
@@ -190,7 +190,20 @@ class ScioInjector extends SyntheticMembersInjector {
         c.contains("extends _root_.com.spotify.scio.avro.types.AvroType.HasAvroAnnotation"))
       .map(_.split("[()]"))
       .map(_.filter(_.contains(" : "))) // get only parameter part
-      .map(_.flatMap(_.split(","))) // get individual parameter
```

Comment:
This code slowly becomes complicated, could you refactor this? Initially this was just a small workaround, but this slowly gets more logic, would be worth to add tests here too.